### ISSUE TITLE
Parent Horizontal Scroll When Dragging Items Doesn't Work Well

### ISSUE THREAD

**Issue by ND56** [formkit/drag-and-drop#21]
First, I want to say that I love the library! 

Onward to the bug-report: I noticed that the horizontal scroll seems to be _very_ touchy when dragging the items. It seems like you need to pause at the edge of the parent for the horizontal scroll to work---and if you go past that edge, the scroll isn't registered.

For example, this is a video of me reproducing the issue in your docs after I resized one of your parent containers and set it to horizontal scroll:

https://github.com/formkit/drag-and-drop/assets/32069832/da19b274-e830-499c-8eb5-02c99e731a49

And then you can see it a bit more pronounced in my own app. It's helpful to see the comparison here where the bottom row is your library and the top row is the react-beautiful-dnd library (where you can see the horizontal scroll is performing more as I would expect).

https://github.com/formkit/drag-and-drop/assets/32069832/00da6ada-8ad5-4c42-a6da-bcf3be6d15ad

Very curious to hear what y'all think about this.

Keep up the good work!

-Nick

**Comment by ND56** [comment]
If it's at all helpful for you, I noticed that---unlike the react-beautiful-dnd library, where the horizontal scroll issue does not occur---the base react-dnd library also has the exact same horizontal scroll issue that your library has.

If you're curious, I reproduced the issue in that library as well in this sandbox: [link](https://codesandbox.io/p/sandbox/fork-of-react-dnd-list-drag-and-drop-sandbox-mpq5g6?file=%2Fsrc%2FCard.js%3A12%2C18).

More importantly, though, I figured I would share that someone actually did already create a fix for the issue in that library, which can be seen here: https://github.com/TechStark/react-dnd-scrolling.

When I leveraged that supplementary package, it did resolve the issue for the react-dnd library.

Figured I would share the solution for that library in case it's helpful for you as you address it in your library!

**Comment by sashamilenkovic** [comment]
@ND56 Really appreciate you digging into this. Will take a look at the linked fix. We're also aiming to make scroll (both vertically and horizontally) work smoothly.

**Comment by sashamilenkovic** [comment]
@ND56 Hey! I just released v0.0.26 if you wanted to take a look. Horizontal scrolling should be working a little bit better. I'll leave this open and wait for your feedback.

**Comment by sashamilenkovic** [comment]
@ND56 Appreciate the detailed feedback and reproduction. Will address each of these issues you brought up.

**Comment by ND56** [comment]
Ah sorry @sashamilenkovic!

I just updated the permissions to be public. Here's another link for good measure: [sandbox](https://codesandbox.io/p/sandbox/formkit-nk6h68?file=%2Fsrc%2FContainer.tsx%3A13%2C4&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clt5y5nft00063b6k7xmmbh65%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clt5y5nfs00023b6kqwrfzzle%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clt5y5nfs00033b6kj7z10n08%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clt5y5nfs00053b6kwdqa2y3n%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B66.33858018104367%252C33.66141981895633%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clt5y5nfs00023b6kqwrfzzle%2522%253A%257B%2522id%2522%253A%2522clt5y5nfs00023b6kqwrfzzle%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cltm331h600023b6kcbj7sus6%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A13%252C%2522startColumn%2522%253A4%252C%2522endLineNumber%2522%253A13%252C%2522endColumn%2522%253A4%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FContainer.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522cltm3vvxa00023b6keqhol3p1%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A68%252C%2522startColumn%2522%253A13%252C%2522endLineNumber%2522%253A68%252C%2522endColumn%2522%253A13%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FContainer.scss%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522cltm331h600023b6kcbj7sus6%2522%257D%252C%2522clt5y5nfs00053b6kwdqa2y3n%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clt5y5nfs00043b6kkby2eide%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clt5y5nfs00053b6kwdqa2y3n%2522%252C%2522activeTabId%2522%253A%2522clt5y5nfs00043b6kkby2eide%2522%257D%252C%2522clt5y5nfs00033b6kj7z10n08%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clt5y5nfs00033b6kj7z10n08%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A10%257D).

**Comment by sashamilenkovic** [comment]
@ND56 Works now, thank you

**Comment by sashamilenkovic** [comment]
@ND56 "However, one issue worth calling out is that I noticed your fix doesn't seem to play well with your animations plugin, so that might be something you'll eventually want to smooth out." 

Yes! Animations definitely don't look like they're playing well with the programmatic scrolling. Would you actually mind filing a separate issue for that?

**Comment by ND56** [comment]
`Regarding Your First Response | The Modal Fix`

Ah, I saw your fix in the sandbox. I do see it looks like it's working as expected!

I do have an issue still on my end with the bizarre placeholder behavior. I tried replacing my own personal modal with the material UI modal in the sandbox and saw the same issue, so at least I know it's not related to the internal mechanics of my modal. Given that it's not the modal code, I think it must somehow be related to my overall app infrastructure that's not reflected in the sandbox I shared (perhaps some kind of excessive re-rendering that's going on that's interfering with the drag and drop behavior).

Definitely seems like a _me_ problem at this point and less of a _you_ problem, so I think it seems like closing this out makes sense (though I will say it's not an issue that I saw before the upgrade). 

`Regarding Your Second Response | The Animations Issue`

Of course, I'll go ahead and do that!
